cmd/govim/buffer_events.go

```diff
@@ -65,6 +77,29 @@ func (v *vimstate) bufReadPost(args ...json.RawMessage) error {
 	return v.handleBufferEvent(nb)
 }
 
+func (v *vimstate) bufQuickFixCmdPre(args ...json.RawMessage) error {
+	v.vimgrepPendingBufs = make(map[int]*types.Buffer)
+	return nil
+}
+
+func (v *vimstate) bufQuickFixCmdPost(args ...json.RawMessage) error {
+	defer func() { v.vimgrepPendingBufs = nil }()
+
+	v.quickfixIsDiagnostics = v.quickfixIsDiagnostics && len(v.vimgrepPendingBufs) == 0
+
+	// Vim versions older than 8.2.2185 did not notify when buffers opened during vimgrep
+	// closed so we need to explicitly check if any of the buffers are still open.
+	for _, b := range v.vimgrepPendingBufs {
```

Comment:
Nit: is this worth doing in a batch? Not sure how expensive this could become.

Reply:
I did consider that but I don't think that it is worth doing. We expect it to be rather few matches, and the additional overhead for a single channel call is probably tiny compared to all the things happening when a buffer is added to govim. It would require two loops as `v.addBuffer(...)` do a channel calls as well that can't be batched. I'll leave it as is for now.